src/Controller/VoucherFundingMerchantController.php

```diff
@@ -17,4 +21,15 @@ public function __construct(VoucherFundingMerchantService $voucherFundingService
     {
         $this->voucherFundingService = $voucherFundingService;
     }
+
+    /**
+     * @Route("/voucher-funding-merchant-api/v1/redeem/voucher", name="voucher-funding-merchant-api.action.redeem-voucher", methods={"POST"})
+     */
+    public function redeemVoucher(Request $request, Context $context): JsonResponse
+    {
```

Comment:
context should change SalesChannelContext $context

add more code for check CustomerNotLoggedInException using SalesChannelContextExtension::extract($context);